import.go

```diff
@@ -44,19 +65,97 @@ func newImporter(tree *MutableTree, version int64) (*Importer, error) {
 		return nil, errors.New("tree must be empty")
 	}
 
-	return &Importer{
-		tree:    tree,
-		version: version,
-		batch:   tree.ndb.db.NewBatch(),
-		stack:   make([]*Node, 0, 8),
-	}, nil
+	importer := &Importer{
+		tree:             tree,
+		version:          version,
+		batch:            tree.ndb.db.NewBatch(),
+		stack:            make([]*Node, 0, 8),
+		batchMtx:         sync.RWMutex{},
+		desiredBatchSize: defaultDesiredBatchSize,
+		maxBatchSize:     defaultMaxBatchSize,
+		chNodeData:       make(chan NodeData, 2*defaultDesiredBatchSize),
+		chNodeDataWg:     sync.WaitGroup{},
+		chBatch:          make(chan db.Batch, 1),
+		chBatchWg:        sync.WaitGroup{},
+		chError:          make(chan error, 1),
+		allChannelClosed: false,
```

Comment:
Without digging into details, this seems like a lot of state to keep 2 goroutines in check. See also my PR comment.

Reply:
Moved them to another struct called ChannelConfig